internal/testutil/acc/atlas.go

```diff
@@ -0,0 +1,112 @@
+package acc
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"runtime"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/atlas-sdk/v20231115007/admin"
+)
+
+// TestMainExecution must be called from TestMain in the test package if ProjectIDExecution is going to be used.
+func TestMainExecution(m *testing.M) {
+	atlasInfo.init = true
+	atlasInfo.resourceName = resourceName()
+
+	exitCode := m.Run()
+
+	if atlasInfo.needsDeletion {
+		fmt.Printf("Deleting execution project: %s, resource: %s\n", atlasInfo.projectName, atlasInfo.resourceName)
+		deleteProject(atlasInfo.projectID)
+	}
+
+	os.Exit(exitCode)
+}
+
+// ProjectIDExecution returns a project id created for the execution of the resource tests.
+func ProjectIDExecution(tb testing.TB) string {
+	tb.Helper()
+	SkipInUnitTest(tb)
+	require.True(tb, atlasInfo.init, "TestMainExecution must called to be able to use ProjectIDExecution")
+
+	atlasInfo.mu.Lock()
+	defer atlasInfo.mu.Unlock()
+
+	// lazy creation so it's only done if really needed
+	if atlasInfo.projectName == "" {
+		var globalName, globalID string
+		if atlasInfo.resourceName != "" {
+			globalName = (prefixProjectKeep + "-" + atlasInfo.resourceName)[:projectNameMaxLen]
+			globalID = projectID(globalName)
+		}
+
+		if globalID == "" {
+			atlasInfo.projectName = RandomProjectName()
+			tb.Logf("Creating execution project: %s, resource: %s, global project (not found): %s\n", atlasInfo.projectName, atlasInfo.resourceName, globalName)
+			atlasInfo.projectID = createProject(tb, atlasInfo.projectName)
+			atlasInfo.needsDeletion = true
```

Comment:
this flags that we created a project for the test execution and we'll need to delete it after all tests finish